tests/mtouch/ToolTasksBinPathTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Diagnostics;
+using NUnit.Framework;
+using Xamarin.Tests;
+
+namespace AppleSdkSettings {
+
+	public static class XcodeVersion {
+		public static int Major = Version.Parse (Configuration.XcodeVersion).Major;
+	}
+}
+
+namespace Xamarin.MacDev.Tasks {
+
+	public abstract class MetalTaskBase {
+
+		protected abstract string OperatingSystem {
+			get;
+		}
+
+		protected abstract string DevicePlatformBinDir {
+			get;
+		}
+
+		protected string SdkDevPath {
+			get { return ""; }
+		}
+	}
+
+	public abstract class MetalLibTaskBase {
+
+		protected abstract string DevicePlatformBinDir {
+			get;
+		}
+
+		protected string SdkDevPath {
+			get { return ""; }
+		}
+	}
+}
+
+namespace Xamarin.Mac.Tasks {
+
+	public class MetalPoker : Metal {
+
+		public new string DevicePlatformBinDir {
+			get { return base.DevicePlatformBinDir; }
+		}
+	}
+
+	public class MetalLibPoker : MetalLib {
+
+		public new string DevicePlatformBinDir {
+			get { return base.DevicePlatformBinDir; }
+		}
+	}
+
+	[TestFixture]
+	public class ToolTasksBinPathTest {
+
+		[SetUp]
+		public void Setup ()
+		{
+			Environment.SetEnvironmentVariable ("DEVELOPER_DIR", Configuration.xcode_root);
+		}
+
+		[Test]
+		public void MetalBinPathTest ()
+		{
+			var metalTask = new MetalPoker ();
+
+			var psi = new ProcessStartInfo ("xcrun") {
+				Arguments = "-f metal",
+				UseShellExecute = false,
+				CreateNoWindow = true,
+				RedirectStandardOutput = true,
+			};
+			var proc = Process.Start (psi);
+
+			string output = proc.StandardOutput.ReadToEnd ();
+			Console.WriteLine (output); // Print the output in case something wrong happens when executing the command
```

Comment:
Do we get anything in StandardError?

Reply:
Probably but I'm not sure it's critical to get it. The standard output tells you what happened. The test will fail also.